src/ucp/wireup/wireup_ep.c

```diff
@@ -420,34 +420,29 @@ ucs_status_t ucp_wireup_ep_connect(uct_ep_h uct_ep, const ucp_ep_params_t *param
     return status;
 }
 
-static ucs_status_t ucp_wireup_ep_pack_always_supported_rscs(ucp_worker_h worker,
-                                                             const char *dev_name,
-                                                             uint64_t *tl_bitmap_p,
-                                                             ucp_address_t **address_p,
-                                                             size_t *address_length_p)
+static ucs_status_t ucp_wireup_ep_pack_sockaddr_aux_tls(ucp_worker_h worker,
+                                                        const char *dev_name,
+                                                        uint64_t *tl_bitmap_p,
+                                                        ucp_address_t **address_p,
+                                                        size_t *address_length_p)
 {
     ucp_context_h context = worker->context;
-    int tl_id, tl, found_supported_tl = 0;
+    int tl_id, found_supported_tl = 0;
     ucs_status_t status;
     uint64_t tl_bitmap = 0;
 
-    if (context->config.num_partial_tls == 0) {
+    if (context->config.sockaddr_aux_rscs.num_rscs == 0) {
         ucs_error("no supported transports found in configuration");
         status = UCS_ERR_UNREACHABLE;
         goto out;
     }
 
     /* Find a transport which matches the given dev_name and the user's configuration */
-    for (tl = 0; tl < context->config.num_partial_tls; ++tl) {
-        for (tl_id = 0; tl_id < context->num_tls; ++tl_id) {
-            if ((!strncmp(context->config.partial_tls[tl].tl_name,
-                          context->tl_rscs[tl_id].tl_rsc.tl_name,
-                          UCT_TL_NAME_MAX)) &&
-                (!strncmp(context->tl_rscs[tl_id].tl_rsc.dev_name, dev_name,
-                          UCT_DEVICE_NAME_MAX))) {
-                found_supported_tl = 1;
-                tl_bitmap |= UCS_BIT(tl_id);
-            }
+    ucs_for_each_bit(tl_id, context->config.sockaddr_aux_rscs.bitmap) {
+        if (!strncmp(context->tl_rscs[tl_id].tl_rsc.dev_name, dev_name,
+                     UCT_DEVICE_NAME_MAX)) {
+            found_supported_tl = 1;
```

Comment:
the variable found_supported_tl should be removed. 
seems this is checked already in line 434, no?

Reply:
@yosefe why removed? it is also possible that neither ud nor ud_x are available and we can't pack anything. (also, you asked to remove the check in line 434 by removing this variable)

Reply:
right, so need to keep this variable and just remove the check in line 434, the check in line 449 should be enough